### PR TITLE
API diff between .NET 9 Preview 7 and .NET 9 RC 1

### PR DESCRIPTION
**Pull request by carlossanlop**
Repo area owners:

- ASP.NET - @dotnet/aspnet-api-review 
- WinForms - @merriemcgaw @lonitra @JeremyKuhne
- WPF - @singhashish-wpf and @pchaurasia14
- Libraries - @artl93 @jeffschwMSFT @jeffhandley @karelz @ericstj @stephentoub @SamMonoRT 

Libraries area owners:
- System @dotnet/area-system-runtime @dotnet/area-system-console 
- System.Collections @dotnet/area-system-collections 
- System.Drawing @dotnet/area-system-drawing @lonitra @merriemcgaw @JeremyKuhne 
- System.IO.Compression @dotnet/area-system-io-compression @buyaa-n 
- System.Net @dotnet/ncl 
- System.Reflection.Emit @dotnet/area-system-reflection-emit 
- System.Reflection.Metadata @dotnet/area-system-reflection-metadata 
- System.Runtime.Intrinsics @dotnet/area-system-runtime-intrinsics 
- System.Security @dotnet/area-system-security 

**Known api-diff tool issues**:

If yo usee any of these, please provide a GitHub suggestion in this PR to correct it: 

- The AsmDiff tool sometimes skips adding `{` or `{}` [arcade#10981](https://github.com/dotnet/arcade/issues/10981)(https://github.com/dotnet/arcade/issues/13814) 
- RequiresPreviewAttribute not getting auto added [arcade#14498](https://github.com/dotnet/arcade/issues/14498)
- Changes to `protected internal` visibility should stop showing. https://github.com/dotnet/arcade/issues/14579
- The AsmDiff tool shows APIs with decorating attribute changes as removed and re-added, instead of showing only the API. Please ignore this, it might be confusing but technically it's not incorrect: [arcade#13814]

### CODE REVIEW

File: release-notes/9.0/preview/rc1/api-diff/Microsoft.NETCore.App/9.0-rc1_System.Net.WebSockets.md
Comment (MihaZupan) [inline release-notes/9.0/preview/rc1/api-diff/Microsoft.NETCore.App/9.0-rc1_System.Net.WebSockets.md:1]: System.Net LGTM

**Review by adamsitnik — APPROVED** [review]
`System.Reflection.Metadata` LGTM, thank you @carlossanlop !

**Review by eiriktsarpalis — APPROVED** [review]
System.Collections LGTM

**Review by eiriktsarpalis — COMMENTED** [review]
Shouldn't this also include the JsonElement.GetPropertyCount that was retroactively added to RC1 via https://github.com/dotnet/runtime/pull/106576?

**Comment by carlossanlop** [comment]
> Shouldn't this also include the JsonElement.GetPropertyCount that was retroactively added to RC1 via [dotnet/runtime#106576](https://github.com/dotnet/runtime/pull/106576)?

@eiriktsarpalis That backport https://github.com/dotnet/runtime/pull/106576 was merged into release/9.0 which from the start was pointing to RC2. I don't see any RC1 backport PRs unfortunately.

**Comment by eiriktsarpalis** [comment]
> That backport [dotnet/runtime#106576](https://github.com/dotnet/runtime/pull/106576) was merged into release/9.0 which from the start was pointing to RC2. I don't see any RC1 backport PRs (release/9.0-rc1) unfortunately.

Ah good, before going OOF I was under the impression it would be merged for RC1 but I guess we missed the deadline.

File: release-notes/9.0/preview/rc1/api-diff/Microsoft.NETCore.App/9.0-rc1_System.IO.Compression.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
++    }
++    public enum ZLibCompressionStrategy {
++        Default = 0,
++        Filtered = 1,
++        Fixed = 4,
++        HuffmanOnly = 2,
++        RunLengthEncoding = 3,
++    }
+     public sealed class ZLibStream : Stream {
++        public ZLibStream(Stream stream, ZLibCompressionOptions compressionOptions, bool leaveOpen = false);
```
Comment (buyaa-n) [inline release-notes/9.0/preview/rc1/api-diff/Microsoft.NETCore.App/9.0-rc1_System.IO.Compression.md:31]: System.IO.Compression part LGTM

File: release-notes/9.0/preview/rc1/api-diff/Microsoft.NETCore.App/9.0-rc1_System.Reflection.Emit.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
++        public PropertyBuilder DefineProperty(string name, PropertyAttributes attributes, Type? returnType, Type[]? returnTypeRequiredCustomModifiers, Type[]? returnTypeOptionalCustomModifiers, Type[]? parameterTypes, Type[][]? parameterTypeRequiredCustomModifiers, Type[][]? parameterTypeOptionalCustomModifiers);
+-        protected abstract PropertyBuilder DefinePropertyCore(string name, PropertyAttributes attributes, CallingConventions callingConvention, Type returnType, Type[]? returnTypeRequiredCustomModifiers, Type[]? returnTypeOptionalCustomModifiers, Type[]? parameterTypes, Type[][]? parameterTypeRequiredCustomModifiers, Type[][]? parameterTypeOptionalCustomModifiers);
++        protected abstract PropertyBuilder DefinePropertyCore(string name, PropertyAttributes attributes, CallingConventions callingConvention, Type? returnType, Type[]? returnTypeRequiredCustomModifiers, Type[]? returnTypeOptionalCustomModifiers, Type[]? parameterTypes, Type[][]? parameterTypeRequiredCustomModifiers, Type[][]? parameterTypeOptionalCustomModifiers);
```
Comment (buyaa-n) [inline release-notes/9.0/preview/rc1/api-diff/Microsoft.NETCore.App/9.0-rc1_System.Reflection.Emit.md:15]: LGTM, note that these are just a nullability change for one parameter `Type returnType` => `Type? returnType`

File: release-notes/9.0/preview/rc1/api-diff/Microsoft.NETCore.App/9.0-rc1_System.Runtime.Intrinsics.Arm.md [hunk 3]
```diff
@@ -0,0 +1,179 @@
+# System.Runtime.Intrinsics.Arm
```
Comment (tannergooding) [inline release-notes/9.0/preview/rc1/api-diff/Microsoft.NETCore.App/9.0-rc1_System.Runtime.Intrinsics.Arm.md:1]: System.Runtime.Intrinsics.Arm LGTM

CC. @dotnet/area-system-runtime-intrinsics, @dotnet/arm64-contrib

File: release-notes/9.0/preview/rc1/api-diff/Microsoft.NETCore.App/9.0-rc1_System.Runtime.Intrinsics.X86.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
+# System.Runtime.Intrinsics.X86
```
Comment (tannergooding) [inline release-notes/9.0/preview/rc1/api-diff/Microsoft.NETCore.App/9.0-rc1_System.Runtime.Intrinsics.X86.md:1]: System.Runtime.Intrinsics.X86 LGTM

CC. @dotnet/area-system-runtime-intrinsics

**Review by tannergooding — APPROVED** [review]

**Comment by bartonjs** [comment]
System.Security.* LGTM

**Pull request merged**